### PR TITLE
Draft: Feature/mac performance

### PR DESCRIPTION
**Pull request by ben-styling**
# Work In Progress

**Describe your change**
Performance on OSX is similar to performance on windows without WSL2. This PR adds an env var to easily allow changing to volume mount flags, and adds docker-sync config

**Type of Change**
What types of changes does your code introduce? Put an `x` in all the boxes that apply:
- [x] New feature (non-breaking change which adds functionality).

**Checklist**
Go over all the following points, and put an `x` in all the boxes that apply
- [x] I've read the [Contribution Guide](https://github.com/chrisnharvey/magicLAMP/blob/master/CONTRIBUTING.md).
- [x] I've updated the **documentation**.
- [x] I've targeted the correct branch with this Pull Request.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,4 +3,7 @@ docker-compose.override.yml
 data/*
 !data/.gitkeep
 containers/custom/*
-!containers/custom/.gitkeep
\ No newline at end of file
+!containers/custom/.gitkeep
+.DS_Store
+daemon.log
+daemon.pid
```
Comment (chrisnharvey) [inline .gitignore:9]: What creates these files? Do they need to be ignored?

File: docker-compose-dev.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      - mysql-data-sync:/var/lib/mysql:nocopy
+    environment:
+      MYSQL_ALLOW_EMPTY_PASSWORD: 1
+      MYSQL_ROOT_PASSWORD:
+
+volumes:
+  projects-sync:
+    external: true
+  mysql-data-sync:
+    external: true
```
Comment (chrisnharvey) [inline docker-compose-dev.yml]: Is this file purely for docker-sync? If so, could we rename it to something like `docker-compose.sync.yml`?

File: docker-compose.override.example.yml [hunk 3]
```diff
@@ -9,3 +9,15 @@ services:
...
+
+  mysql:
+    image: mariadb:latest
+
+    volumes:
+      - mysql_data:/var/lib/mysql:rw${VOLUME_FLAGS}
+
+    environment:
+      MYSQL_ALLOW_EMPTY_PASSWORD: 1
+      MYSQL_ROOT_PASSWORD:
```
Comment (chrisnharvey) [inline docker-compose.override.example.yml]: Are these changes required for this PR?

File: docker-compose.yml [hunk 4]
```diff
@@ -19,6 +19,9 @@ services:
   postgres:
     image: chrisnharvey/magiclamp-postgres:${MAGICLAMP_VERSION:-1.5}
 
+    volumes:
+      - pg_data:/var/lib/postgresql/data/pgdata:rw${VOLUME_FLAGS}
+
```
Comment (chrisnharvey) [inline docker-compose.yml:24]: I think adding this line actually solves a bug, nice catch! ;)

Would be cool if you could cherry-pick this into a new PR that targets `master`.

File: docker-sync.yml [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  compose-dev-file-path: 'docker-compose-dev.yml'
+
+syncs:
+
+  projects-sync:
+    src: '${PROJECTS_DIR}/'
+
+    sync_userid: '${USER_UID}'
+
+    # 2do: change the volume flags var and usage to allow use here.
```
Comment (chrisnharvey) [inline docker-sync.yml:14]: Can you remove this TODO comment? Or replace it with something else?

File: docs/troubleshooting/performance-on-osx.md [hunk 6]
```diff
@@ -0,0 +1,27 @@
+# Performance on OSX
```
Comment (chrisnharvey) [inline docs/troubleshooting/performance-on-osx.md]: Could we change this to `macOS` instead of `OSX`?

**Review by chrisnharvey — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (ben-styling) [reply .gitignore:9]: Correct me if I'm wrong - still really new to a lot of this dev ops-y stuff -  but this should be a generated file for when a daemon is running to store it's ID so it can't be run twice.

This must be created by either docker sync or unison.

I added to gitignore simply because it's a generated file. Not sure if this is correct or not, though.

File: docker-compose.override.example.yml (hunk 3 above)
Comment (ben-styling) [reply docker-compose.override.example.yml]: No it's not. I'll remove and bring into another PR.

**Comment by ben-styling** [comment]
FYI: Closing as this is a very old PR and probably out of date.
